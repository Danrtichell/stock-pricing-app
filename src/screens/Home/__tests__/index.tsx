import React from 'react';
import {Provider} from 'react-redux';
import renderer from 'react-test-renderer';
import {Home} from '..';
import configureStore from '../../../configureStore';

const renderComponent = () => {
  const store = configureStore({stock: {data: [], stocksInfo: {}}});
  return renderer.create(
    <Provider store={store}>
      <Home />
    </Provider>,
  );
};

describe('<Home />', () => {
  it('should render correctly', () => {
    const component = renderComponent();
    expect(component.toJSON()).toMatchSnapshot();
  });
});
