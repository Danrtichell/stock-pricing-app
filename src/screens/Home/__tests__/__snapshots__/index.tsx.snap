// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should render correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "alignItems": "flex-start",
      "flex": 1,
      "flexDirection": "row",
      "height": "100vh",
      "justifyContent": "center",
      "marginHorizontal": 20,
      "marginTop": 50,
    }
  }
>
  <View
    style={
      Object {
        "flex": 2,
        "flexDirection": "column",
        "marginRight": 20,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 30,
          }
        }
      >
        Logs
      </Text>
      <View
        accessibilityRole="button"
        accessibilityState={Object {}}
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            Pause Log
          </Text>
        </View>
      </View>
    </View>
    <RCTScrollView
      scrollEnabled={true}
      style={
        Object {
          "borderColor": "black",
          "borderStyle": "solid",
          "borderWidth": 1,
          "marginTop": 20,
          "maxHeight": 600,
          "minHeight": 300,
          "padding": 20,
        }
      }
    >
      <View />
    </RCTScrollView>
  </View>
  <View
    style={
      Object {
        "flex": 4,
        "flexDirection": "column",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 30,
        }
      }
    >
      Summary
    </Text>
    <View
      style={
        Object {
          "borderColor": "black",
          "borderStyle": "solid",
          "borderWidth": 1,
          "height": 300,
          "marginTop": 20,
          "padding": 20,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "column",
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 14,
                  "fontWeight": "bold",
                }
              }
            >
              Stock
            </Text>
          </View>
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 14,
                  "fontWeight": "bold",
                }
              }
            >
              Starting
            </Text>
          </View>
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 14,
                  "fontWeight": "bold",
                }
              }
            >
              Lowest
            </Text>
          </View>
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 14,
                  "fontWeight": "bold",
                }
              }
            >
              Highest
            </Text>
          </View>
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 14,
                  "fontWeight": "bold",
                }
              }
            >
              Current
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
